src/to-have-text-content.ts

```diff
@@ -7,15 +7,14 @@ function collectNormalizedText(element: ReactTestInstance) {
   return normalize(childrenText);
 }
 
-function collectChildrenText(element: string | ReactTestInstance): string[] {
+function collectChildrenText(element: ReactTestInstance | string): string[] {
   if (typeof element === 'string') return [element];
-  if (!element || !element.children) return [''];
+  if (!element || !element.children) return [];
 
-  let result: string[] = [];
-
-  for (const child of element.children) {
-    result = result.concat(collectChildrenText(child));
-  }
+  const result: string[] = [];
+  element.children.forEach((child) => {
```

Comment:
out of curiosity, why the forEach instead of the for of?

Reply:
No particular reason beside subjectively slightly easier to read.